File: src/main/kotlin/br/com/caethas/atto/atto/modules/membro/enums/StatusAssociado.kt
package br.com.caethas.atto.atto.modules.membro.enums

import com.fasterxml.jackson.annotation.JsonCreator

enum class StatusAssociado(val value: Int, val label: String) {
    ATIVO(0, "Ativo"),
    INATIVO(1, "Inativo"),
    DESLIGADO(2, "Desligado"),
    INADIMPLENTE(3, "Inadimplente");

    companion object {
        fun fromValue(value: Int): StatusAssociado =
            entries.find { it.value == value }
                ?: throw IllegalArgumentException("Status desconhecido: $value")
    }
}

File: lib/src/modules/membro/enums/status_associado.dart
enum StatusAssociado {
  ativo(0, 'Ativo'),
  inativo(1, 'Inativo'),
  desligado(2, 'Desligado'),
  inadimplente(3, 'Inadimplente');

  final int value;
  final String label;

  const StatusAssociado(this.value, this.label);

  static StatusAssociado fromValue(int value) {
    return StatusAssociado.values.firstWhere(
          (e) => e.value == value,
      orElse: () => StatusAssociado.inativo,
    );
  }

  Map<String, dynamic> toJson() => {'value': value, 'label': label};
}